### PR TITLE
Add Hyperledger Indy Node blueprint

### PR DESCRIPTION
**Pull request by KatsuyaMatsuoka**
### What does this PR do?

#33 

Add Hyperledger Indy blueprint, CDK assets and docs.
It creates Indy Trustee and Steward Nodes.

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/aws-samples/aws-blockchain-node-runners/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

### Motivation

To support Indy. There is [community's document about AWS](https://github.com/hyperledger/indy-node/blob/main/docs/source/install-docs/AWS-NodeInstall-20.04.md), but there is no IaC asset.

### More

- [ ] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Mandatory for new node blueprints. Yes, I have added usage example to the `README.md` file in my blueprint folder
- [ ] Mandatory for new node blueprints. Yes, I have implemented automated tests for all stacks in my blueprint and they pass
- [ ] Mandatory for new node blueprints. Yes, I have added a reference to my `README.md` file to `website/docs` section for this feature
- [ ] Yes, I have set up and ran all [pre-merge quality control tools](./docs/pre-merge-tools.md) on my local machine and they don't show warnings.

### For Moderators

- [ ] The tests for all current blueprints successfully complete before merge?
- [ ] Mandatory for new node blueprints. All [pre-merge quality control tools](./docs/pre-merge-tools.md) and `cdk-nag` tools don't show warnings?
- [ ] Mandatory for new node blueprints. The deployment test works on blank AWS account according to instructions in the `README.md` before merge?
- [ ] Mandatory for new node blueprints. The website builds without errors?

### CODE REVIEW

File: lib/indy/README.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+```bash
+sudo init_indy_node <ALIAS> <NODE_IP> 9701 <CLIENT_IP> 9702 <SEED>
+```
+
+**NOTE:** Here, Steward represents Validator Node ([reference information](https://github.com/pSchlarb/indy-node/blob/documentationUpdate/docs/source/installation-and-configuration.md#32-validator-node-installation)).
+
+#### Generating Genesis files
+
+1. Download each sheet (Stewards, Trustees) containing the information generated in the steps so far
```
Comment (frbrkoala) [inline lib/indy/README.md:94]: Need guidance where I should go to download the files from? I've got stuck on this step and could not continue.

**Review by frbrkoala — CHANGES_REQUESTED** [review]
Need your help to make changes to the CDK and add some clarity to the instructions in the README. Got stuck on step "Generating Genesis Files" and could not continue because could not figure out where to download the files from.

Overall, it looks like the current instructions include a lot of manual steps. Is it possible to script them up so that the network will get up and running all together automatically?

File: lib/indy/lib/constructs/indy-node-instance.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      instanceType: ec2.InstanceType.of(ec2.InstanceClass.T3, ec2.InstanceSize.LARGE),
+      machineImage: ec2.MachineImage.fromSsmParameter(
+        '/aws/service/canonical/ubuntu/server/focal/stable/current/amd64/hvm/ebs-gp2/ami-id'
+      ),
+      ssmSessionPermissions: true,
+      userData: ec2.UserData.custom(readFileSync("./lib/assets/user-data/steward.sh", "base64")),
+      blockDevices: [{
+        deviceName: "/dev/sda1",
+        volume: ec2.BlockDeviceVolume.ebs(250, {
+          volumeType: ec2.EbsDeviceVolumeType.STANDARD,
```
[the remark this answers is not in the document]
Comment (KatsuyaMatsuoka) [reply lib/indy/lib/constructs/indy-node-instance.ts:46]: According to the community doc I referred to, the magnetic standard is recommended for cost reduction, so I adopted this one, but I think there is no problem with GP3.

https://github.com/hyperledger/indy-node/blob/2b61cd8340886945d550eb8a5218ab3d71223668/docs/source/install-docs/AWS-NodeInstall-20.04.md?plain=1#L45

File: lib/indy/lib/constructs/indy-node-instance.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    const nodeNic: ec2.CfnInstance.NetworkInterfaceProperty = {
+      deviceIndex: "1",
+      groupSet: [nodeSG.securityGroupId],
+      subnetId: vpc.privateSubnets[0].subnetId,
+      description: 'Node NIC',
+    };
+
+    const instance = new ec2.Instance(this, 'Instance', {
+      vpc: vpc,
+      instanceType: ec2.InstanceType.of(ec2.InstanceClass.T3, ec2.InstanceSize.LARGE),
```
[the remark this answers is not in the document]
Comment (KatsuyaMatsuoka) [reply lib/indy/lib/constructs/indy-node-instance.ts:37]: I chose T3.large based on community doc, but I think M5 is better when considering use in production environments.

https://github.com/hyperledger/indy-node/blob/2b61cd8340886945d550eb8a5218ab3d71223668/docs/source/install-docs/AWS-NodeInstall-20.04.md?plain=1#L21

**Pull request merged**